src/schemaValidation/validateSchema.ts

```diff
@@ -0,0 +1,169 @@
+import { CliError } from "../params";
+import { ReleaseFileType } from "../types";
+import Ajv, { ErrorObject } from "ajv";
+import ajvErrors from "ajv-errors";
+// Schemas
+import manifestSchema from "./schemas/manifest.schema.json";
+import composeSchema from "./schemas/compose.schema.json";
+import setupWizardSchema from "./schemas/setup-wizard.schema.json";
+import { SetupWizard } from "../releaseFiles/setupWizard/types";
+import { Manifest } from "../releaseFiles/manifest/types";
+import { Compose } from "../releaseFiles/compose/types";
+
+const ajv = new Ajv({
+  allErrors: true,
+  coerceTypes: true,
+  strictSchema: false
+});
+
+ajvErrors(ajv);
+
+console.log("Validating manifest schema");
+const manifestValidate = ajv.compile(manifestSchema);
+console.log("Validating setupwizard schema");
+const setupWizardValidate = ajv.compile(setupWizardSchema);
+console.log("Validating compose schema");
+const composeValidate = ajv.compile(composeSchema); // compose schema https://raw.githubusercontent.com/compose-spec/compose-spec/master/schema/compose-spec.json
+
+/**
+ * Validates a manifest syncronously. Does NOT throw.
+ * @param manifest
+ * @returns = {
+ *   valid: false|true
+ *   errors: [
+ *     "manifest should have required property 'description'",
+ *     "manifest.image.path should be an non-empty string",
+ *   ]
+ * }
+ */
+export function validateSchema(
+  releaseFile: Manifest | Compose | SetupWizard
+): void {
+  const { valid, errors } = validateReleaseFile(releaseFile);
+  if (valid) return;
+
+  // If not valid, print errors and stop execution
+  throw new CliError(
+    `Invalid ${releaseFile}: \n${errors.map(msg => `  - ${msg}`).join("\n")}`
+  );
+}
+
+/**
+ * Validates a manifest syncronously. Does NOT throw.
+ * @param manifest
+ * @returns = {
+ *   valid: false|true
+ *   errors: [
+ *     "manifest should have required property 'description'",
+ *     "manifest.image.path should be an non-empty string",
+ *   ]
+ * }
+ */
+export function validateReleaseFile(
```

Comment:
What's the point of multplexing arguments here with `inferReleaseFileType`? Why not just do:

```ts
function validateManifestSchema(manifest: Manifest)
function validateComposeSchema(compose: Compose)
function validateSetupWizardSchema(setupWizard: SetupWizard)